src/screens/Messages/Conversation/MessagesList.tsx

```diff
@@ -289,15 +313,110 @@ export function MessagesList({
           onScrollToIndexFailed={onScrollToIndexFailed}
           scrollEventThrottle={100}
           ListHeaderComponent={
-            <MaybeLoader isLoading={convo.isFetchingHistory} />
+            <MaybeLoader isLoading={convoState.isFetchingHistory} />
           }
         />
       </ScrollProvider>
-      <MessageInput
-        onSendMessage={onSendMessage}
-        scrollToEnd={scrollToEndNow}
-      />
+      {!isBlocking && !isBlockedBy ? (
+        <MessageInput
+          onSendMessage={onSendMessage}
+          scrollToEnd={scrollToEndNow}
+        />
+      ) : (
+        <BlockedFooter
+          recipient={recipient}
+          convoId={convoState.convo.id}
+          hasMessages={convoState.items.length > 0}
+          isBlocking={isBlocking}
+          isBlockedBy={isBlockedBy}
+        />
+      )}
       {showNewMessagesPill && <NewMessagesPill />}
     </Animated.View>
   )
 }
+
+function BlockedFooter({
+  recipient,
+  convoId,
+  hasMessages,
+  isBlocking,
+  isBlockedBy,
+}: {
+  recipient: Shadow<AppBskyActorDefs.ProfileViewBasic>
+  convoId: string
+  hasMessages: boolean
+  isBlocking: boolean
+  isBlockedBy: boolean
+}) {
+  const t = useTheme()
+  const {_} = useLingui()
+  const [__, queueUnblock] = useProfileBlockMutationQueue(recipient)
+
+  const leaveConvoControl = useDialogControl()
+  const reportDialogControl = useDialogControl()
+
+  if (!isBlocking && !isBlockedBy) return null
+
+  return (
+    <View style={[hasMessages && a.py_md, a.gap_lg]}>
+      {hasMessages && (
+        <>
+          <Divider />
+        </>
+      )}
+      <Text style={[a.text_md, a.font_bold, a.text_center]}>
+        {_(msg`You have blocked this user`)}
+      </Text>
+
+      <View style={[a.flex_row, a.justify_between, a.gap_lg, a.px_md]}>
+        <Button
+          label={_(msg`Delete`)}
+          color="secondary"
+          variant="solid"
+          size="small"
+          style={[a.flex_1]}
+          onPress={leaveConvoControl.open}>
+          <ButtonText style={{color: t.palette.negative_500}}>
+            <Trans>Delete</Trans>
```

Comment:
Should we keep this consistent and say "Leave chat"?

Reply:
Sure. I think I'll need to change the layout of the buttons a bit on native for this but like the idea.